src/Author/Command/RefreshMapfileCommand.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace GisClient\Author\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+
+class RefreshMapfileCommand extends Command
+{
+    protected function configure()
+    {
+        $this->setName("gisclient:refresh-mapfile")
+            ->setDescription("Refreshes a mapfile")
+            ->setHelp("This command allows you to refresh a mapfile.")
+            ->addArgument(
+                "project",
+                InputArgument::OPTIONAL,
+                "Which project is the mapset a part of?",
+                "all"
+            )
+            ->addArgument(
+                "mapset",
+                InputArgument::OPTIONAL,
+                "What's the name of the mapset to refresh? (Defaults to 'all' when project equals 'all')",
+                "all"
+            )
+            ->addOption(
+                "temporary",
+                "t",
+                InputOption::VALUE_NONE,
+                "Only refresh temporary mapfile(s)"
+            )
+            ->addOption(
+                "layer-mapfile",
+                "l",
+                InputOption::VALUE_NONE,
+                "Additionaly create/refresh mapfile for each layer"
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```

Comment:
This function should return a exit code different from 0 when a exception occur

Reply:
Uncaught Exception cause the command to exit with error code 1